fri/src/prover/mod.rs

```diff
@@ -229,48 +257,41 @@ where
     /// # Panics
     /// Panics is the prover state is clean (no FRI layers have been build yet).
     pub fn build_proof(&mut self, positions: &[usize]) -> FriProof {
-        assert!(
-            !self.layers.is_empty(),
-            "FRI layers have not been built yet"
-        );
```

Comment:
I'd still keep a similar check here. What we can assert is that remainder should not be empty at this point. If it is, we haven't built FRI layers yet.

Reply:
Makes sense, I included the same assert with the same message again.